### DISCUSSION TITLE
Select multiple features with vl-interaction-select

### DISCUSSION THREAD

**Discussion by acostela** [ghettovoice/vuelayers#400] [Q&A]
Hi I'm starting to use your library and it's amazing. I'm still a bit newbie and I'm trying to select multiple features clicking using the vl-interaction-select component. I'm able to click on one element and select it but I want to override by default behavior so when I click in another feature it's also selected without unselecting the next one. 

I'm able to call functions onSelect and onUnselect but I don't know how to extract the feature from the event. I tried a lot of different things but I always get TypeError Converting Circular structure to JSON and I think it's because when I extract the geometry values I hve a problem with constructor that is including the object itself. How can I extract this?
```

 onSelect (evt) {
                if(this.selectedFeatures.length > 0) {
                    var featureToAdd = {};
                    featureToAdd.id = evt.id_;
                    featureToAdd.geometry = evt.values_;
                    this.selectedFeatures.push(featureToAdd);
                }
```

I'm using it like that, where selectedFeatures is an array with the selected features. Thanks a lot!

It was a problem trying to stringify the feature itself in the HTML so now it's solved. I don't know how to delete this discussion sorry ;)

**Comment by ghettovoice** [comment]
Hello @acostela ,
you need configure vl-interaction-select with custom toggle condition. By default you can add feature to selection by holding shift and click on it.
With custom toggle logic example https://jsfiddle.net/ghettovoice/qa8p913o/189/

**Reply by acostela** [reply]
Thanks a lot! I did it in a different way but this one is much cleaner so I will refactor my code. Thanks a lot for your help :)
